src/tests/ascii/test_ascii.f90

```diff
@@ -0,0 +1,508 @@
+program test_ascii
+
+    use stdlib_experimental_error, only: assert
+    use stdlib_experimental_ascii
```

Comment:
Can you please explicitly import the symbols that are being used? I think we should follow that approach of "explicit imports", as in Python.

Reply:
(The reason I noticed this is that I came here to see what the public API actually is. As it is a bit hard to immediately see from the module itself, because there is not a single `public` line, but rather quite a few symbols are decorated with `public`.)

Reply:
I think we should have `private` declared at the top of all modules, and below one or more lists of `public :: ...`

Reply:
@zbeekman, I agree, it makes it very easy to see what the public symbols are.